File: Player/src/main/java/ru/testsimpleapps/coloraudioplayer/ui/animation/OnTranslationAnimation.java
package ru.testsimpleapps.coloraudioplayer.ui.animation;


import android.graphics.Rect;
import android.view.View;
import android.view.animation.Animation;
import android.view.animation.LinearInterpolator;
import android.view.animation.TranslateAnimation;

public class OnTranslationAnimation extends BaseAnimationListener {

    private static final int DEFAULT_DURATION = 200;
    private boolean mIsTranslation = false;
    private int mDuration = DEFAULT_DURATION;

    public OnTranslationAnimation(final View view, final int duration) {
        super(view);
        mDuration = duration;
    }

    @Override
    public void onAnimationEnd(Animation animation) {
        super.onAnimationEnd(animation);
        if (mIsTranslation) {
            mView.setVisibility(View.INVISIBLE);
        } else {
            mView.setVisibility(View.VISIBLE);
        }
    }

    public void animate(final boolean isHide) {
        if ((isHide && mView.getVisibility() == View.VISIBLE) ||
                (!isHide && mView.getVisibility() == View.INVISIBLE)) {

            mIsTranslation = isHide;

            // Get view position
            final Rect rect = new Rect();
            mView.getLocalVisibleRect(rect);

            final int height = mView.getHeight();
            float fromTop;
            float toTop;

            // Get new coordinates
            if (isHide) {
                fromTop = rect.top;
                toTop = rect.top - height;
            } else {
                fromTop = rect.top - height;
                toTop = rect.top;
            }

            // Animate transition
            final TranslateAnimation translateAnimation = new TranslateAnimation(rect.left, rect.left, fromTop, toTop);
            translateAnimation.setDuration(mDuration);
            translateAnimation.setInterpolator(new LinearInterpolator());
            translateAnimation.setAnimationListener(this);
            mView.startAnimation(translateAnimation);
        }
    }

}

File: player/src/main/java/ru/testsimpleapps/coloraudioplayer/ui/animation/OnTranslationAnimation.kt
package ru.testsimpleapps.coloraudioplayer.ui.animation


import android.graphics.Rect
import android.view.View
import android.view.animation.Animation
import android.view.animation.LinearInterpolator
import android.view.animation.TranslateAnimation

class OnTranslationAnimation(view: View, duration: Int) : BaseAnimationListener(view) {
    private var mIsTranslation = false
    private var mDuration = DEFAULT_DURATION

    init {
        mDuration = duration
    }

    override fun onAnimationEnd(animation: Animation) {
        super.onAnimationEnd(animation)
        if (mIsTranslation) {
            mView.visibility = View.INVISIBLE
        } else {
            mView.visibility = View.VISIBLE
        }
    }

    fun animate(isHide: Boolean) {
        if (isHide && mView.visibility == View.VISIBLE || !isHide && mView.visibility == View.INVISIBLE) {
            mIsTranslation = isHide

            // Get view position
            val rect = Rect()
            mView.getLocalVisibleRect(rect)

            val height = mView.height
            val fromTop: Float
            val toTop: Float

            // Get new coordinates
            if (isHide) {
                fromTop = rect.top.toFloat()
                toTop = (rect.top - height).toFloat()
            } else {
                fromTop = (rect.top - height).toFloat()
                toTop = rect.top.toFloat()
            }

            // Animate transition
            val translateAnimation = TranslateAnimation(rect.left.toFloat(), rect.left.toFloat(), fromTop, toTop)
            translateAnimation.duration = mDuration.toLong()
            translateAnimation.interpolator = LinearInterpolator()
            translateAnimation.setAnimationListener(this)
            mView.startAnimation(translateAnimation)
        }
    }

    companion object {

        val DEFAULT_DURATION = 200
    }

}
